client/app/bundles/course/assessment/submission/actions/index.js

```diff
@@ -484,15 +485,70 @@ export function saveGrade(submissionId, grades, exp, published) {
   };
 
   return (dispatch) => {
-    dispatch({ type: actionTypes.SAVE_GRADE_REQUEST });
+    dispatch({ type: actionTypes.SAVE_ALL_GRADE_REQUEST });
 
     return CourseAPI.assessment.submissions
       .update(submissionId, payload)
       .then((response) => response.data)
       .then((data) => {
-        dispatch({ type: actionTypes.SAVE_GRADE_SUCCESS, payload: data });
+        dispatch({ type: actionTypes.SAVE_ALL_GRADE_SUCCESS, payload: data });
         dispatch(setNotification(translations.updateSuccess));
       })
+      .catch((error) => {
+        dispatch({ type: actionTypes.SAVE_ALL_GRADE_FAILURE });
+        dispatch(
+          setNotification(translations.updateFailure, buildErrorMessage(error)),
+        );
+      });
+  };
+}
+
+export function saveGrade(
+  submissionId,
+  grades,
```

Comment:
Instead of passing an array that only contains 1 element, suggest passing the raw data and build the payload inside. That way, someone who invokes this function does not need to know the specific format to be passed to the payload.